torch/csrc/utils/tensor_new.cpp

```diff
@@ -53,39 +60,51 @@ void maybe_initialize_cuda(const Device device) {
   }
 }
 
-Tensor dispatch_zeros(const Type& type, const ScalarType scalar_type, optional<Device> device, IntArrayRef sizes) {
-  maybe_initialize_cuda(type);
+Tensor dispatch_zeros(c10::TensorTypeId type_id, at::ScalarType scalar_type, optional<Device> device, IntArrayRef sizes) {
+  maybe_initialize_cuda(type_id);
   AutoNoGIL no_gil;
-  return torch::zeros(sizes, type.options(scalar_type, std::move(device)));
+  if (device.has_value()) {
```

Comment:
for all of these that condition based on whether the device exists...can we just build that into your `options` function?